### PR TITLE
Dev/11: Add whitelist support to variable naming style check

### LINKED ISSUE #11: Change variable naming style rule
# Description

Keep current rules but
1. [x] reject single char name by default
2. [x] reads another whitelist token array to accept these token

### Questions

* How to receive tokens?
* If whitelist is for token and we enable `i`, `j`, `k`:
    * `i_j_k` or `jIK` are valid.
* If whitelist is for variable:
    * `aGoodVariable` will not be valid since `a` is rejected.

# Acceptance Criteria

### PR DESCRIPTION
**Pull request by edisonhello**
Resolves #11.

### CODE REVIEW

File: src/main/python/code_style/rules/naming_style.py [hunk 1]
```diff
@@ -26,18 +28,24 @@ def split_by_uppercase(x, y):
...
+    if variable_name in config.variable_whitelist.split(','):
+        return True
+
     words = split_variable_name(variable_name)
-    return all([is_word(word) for word in words])
+    return all([is_word(word) and (not config.disable_single_character_word or len(word) > 1) for word in words])
+
 
-def naming_style_check(xml):
+def naming_style_check(xml, config):
```
Comment (Johnny850807) [inline src/main/python/code_style/rules/naming_style.py:48]: `config: Config`

File: src/main/python/code_style/rules/naming_style.py [hunk 2]
```diff
@@ -46,7 +54,7 @@ def naming_style_check(xml):
         variable_attr = variable_node.attrib
         variable_token_node = tokenlist_node.find(f'./token[@id=\'{variable_attr["nameToken"]}\']')
         variable_name = variable_token_node.attrib['str']
-        if not is_legal_name(variable_name):
+        if not is_legal_name(variable_name, config):
```
Comment (Johnny850807) [inline src/main/python/code_style/rules/naming_style.py:57]: `config: Config`

File: src/main/python/code_style/rules/naming_style.py [hunk 3]
```diff
@@ -26,18 +28,24 @@ def split_by_uppercase(x, y):
...
 
-def is_legal_name(variable_name):
+
+def is_legal_name(variable_name, config):
+    if variable_name in config.variable_whitelist.split(','):
+        return True
+
     words = split_variable_name(variable_name)
-    return all([is_word(word) for word in words])
+    return all([is_word(word) and (not config.disable_single_character_word or len(word) > 1) for word in words])
```
Comment (Johnny850807) [inline src/main/python/code_style/rules/naming_style.py:45]: `return all([is_legal_word(word) for word in words])`

where 
```python
def is_legal_word(word: str, config: Config) -> bool:
    return is_word(word) and (not config.disable_single_character_word or len(word) > 1)
```

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done, ... and please add as many type checkings as possible.

File: src/main/python/code_style/Config.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Config:
+    def __init__(self, args: dict):
```
Comment (Johnny850807) [inline src/main/python/code_style/Config.py:2]:
```python
def __init__(self, source_root: str, formula: str, disable_single_character_word: str, variable_whitelist: str, **kwargs):
    ...
```

and you can pass the arguments like this
`Config(**args)`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done, thanks!

File: src/main/python/code_style/Config.py (hunk 4 above)
Comment (edisonhello) [reply src/main/python/code_style/Config.py:2]: Seems not working :( 
`TypeError: Config.Config() argument after ** must be a mapping, not Namespace`

Comment (Johnny850807) [reply src/main/python/code_style/Config.py:2]: @edisonhello 

It means the `args` must be a **dict-like** thing,  try `Config(**args.__dict__)`

If this still doesn't work, just pass each of them explicitly.

```python 
def __init__(self, source_root: str, formula: str, disable_single_character_word: str, variable_whitelist: str):
    self.source_root = source_root
    self.formula = formula 
   ... and so on 
```

**Review by Johnny850807 — CHANGES_REQUESTED** [review]

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**